pages/api/post/[id].js

```diff
@@ -0,0 +1,13 @@
+import Post from '../../../src/models/Post';
+
+export default async function handler(req, res) {
+	const { id } = req.query;
+
+	if (req.method === 'DELETE') {
+		const deletedPost = await Post.findByIdAndDelete(id);
+		res.status(200).json({ message: 'post deleted', post: deletedPost });
+	} else {
+		const singleCard = await Post.findById(id);
```

Comment:
Even when a 'POST' or 'PUT' request is made this code is executed. Please add another else if block to check whether the method is 'GET'. (You could use a `switch` statement as well).